middleware/kubernetes/parse.go

```diff
@@ -101,6 +98,5 @@ func (r recordRequest) String() string {
 	s += "." + r.service
 	s += "." + r.namespace
 	s += "." + r.podOrSvc
-	s += "." + r.zone
```

Comment:
sorry, what? You mean each s is a new string and therefor adds GC?
Sure I can change this.